test/support/utils.rb

```diff
@@ -91,6 +89,31 @@ def create_pseudo_terminal(boot_options: "-r debug/run")
 
     private
 
+    # use this to start a debug session with the test program
+    def _start_manual_debug_session(program)
+      print("[Starting a Debug Session with @#{caller.first}]\n")
+      write_temp_file(strip_line_num(program))
+
+      require_relative "../../lib/debug/client"
+
+      socket_path = "#{DEBUGGER__.unix_domain_socket_dir}/test-program-debug-session"
```

Comment:
could you use `DEBUGGER__.create_unix_domain_socket_name` to make a unique socket file?